packages/components/scripts/tests/index.js

```diff
@@ -0,0 +1,64 @@
+import { execSync } from 'child_process';
+import * as jobs from '../jobs/index.js';
+import { getPackageJSONAsObject, getPath } from '../jobs/shared.js';
+
+const componentDir = getPath('../');
+const outDir = getPath('../dist');
+const temporaryOutDir = getPath('../dist.testrunner');
+
+const { version } = getPackageJSONAsObject();
+
+// eslint-disable-next-line no-underscore-dangle
+const __PACKAGE_VERSION__ = JSON.stringify(version.toString());
+
+const synRefreshComponentBundle = () => {
+  const name = 'syn-refresh-component-bundle';
+
+  const rebuildSynergy = async () => {
+    // Create the build in its own directory
+    await jobs.runPrepare(temporaryOutDir);
+    await jobs.runCreateEvents(componentDir);
+    await jobs.runCreateExports(componentDir);
+    await jobs.runTypeScript(temporaryOutDir, './tsconfig.prod.json');
+    await jobs.runEsBuildComponents(temporaryOutDir, __PACKAGE_VERSION__);
+
+    // Finally, remove the dist folder completely
+    // and move the contents of the temporary folder into it
+    await jobs.runPrepare(outDir);
+    execSync(`mv ${temporaryOutDir}/* ${outDir}`);
+  };
+
+  return {
+    name,
+    serverStart: async ({ fileWatcher }) => {
+      // Make sure we always have correct initial data by running the build
+      await rebuildSynergy();
+
+      // Whenever one of our component files has changed,
+      // make sure to recreate our bundle
+      fileWatcher.add('./src/components/**/!(*.(style|test)).ts');
```

Comment:
I would include styles for the tests. If stuff like the visibility has to be tested, this would be part of CSS, shouldn't it?

Reply:
Fixed in latest version